rotkehlchen/chain/evm/decoding/omnibridge/decoder.py

```diff
@@ -0,0 +1,135 @@
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import ChainID, ChecksumEvmAddress
+from rotkehlchen.utils.misc import hex_or_bytes_to_address, hex_or_bytes_to_int
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.user_messages import MessagesAggregator
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class OmnibridgeCommonDecoder(DecoderInterface, abc.ABC):
+
+    def __init__(
+            self,
+            evm_inquirer: 'EvmNodeInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+            deposit_topic: bytes,
+            withdrawal_topic: bytes,
+            bridge_address: ChecksumEvmAddress,
+            source_chain: ChainID,
+            target_chain: ChainID,
+    ) -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.bridge_address = bridge_address
+        self.deposit_topic = deposit_topic
+        self.withdrawal_topic = withdrawal_topic
+        self.source_chain = source_chain
+        self.target_chain = target_chain
+
+    def _decode_bridge_tokens(self, context: DecoderContext) -> DecodingOutput:
+        """Decodes a bridging event for tokens. Either a deposit or a withdrawal."""
+        if context.tx_log.topics[0] == self.deposit_topic:
+            from_address = context.transaction.from_address
+            to_address = from_address  # We have no to_address information
+        elif context.tx_log.topics[0] == self.withdrawal_topic:
+            to_address = hex_or_bytes_to_address(context.tx_log.topics[2])
+            from_address = to_address  # We have no from_address information
+        else:
+            return DEFAULT_DECODING_OUTPUT
+
+        bridged_asset = get_or_create_evm_token(
+            userdb=self.evm_inquirer.database,
+            evm_address=hex_or_bytes_to_address(context.tx_log.topics[1]),
+            chain_id=self.evm_inquirer.chain_id,
+            evm_inquirer=self.evm_inquirer,
+        )
+        amount = asset_normalized_value(
+            amount=hex_or_bytes_to_int(context.tx_log.data[0:32]),
+            asset=bridged_asset,
+        )
+        expected_event_type, new_event_type, from_chain, to_chain, expected_location_label = bridge_prepare_data(  # noqa: E501
+            tx_log=context.tx_log,
+            deposit_topics=(self.deposit_topic,),
+            source_chain=self.source_chain,
+            target_chain=self.target_chain,
+            from_address=from_address,
+            to_address=to_address,
+        )
+
+        for event in context.decoded_events:
+            if bridged_asset == A_WETH and event.asset == A_ETH:
+                if event.event_type == HistoryEventType.RECEIVE:
```

Comment:
here I expect the gas event to fall under the first `if` condition. You can check against the event.amount since you know the bridged WETH amount and WETH is 1:1 with ETH.